lib/kernel/doc/src/gen_sctp.xml

```diff
@@ -79,15 +79,12 @@
       </desc>
     </datatype>
     <datatype>
-      <name name="option"/>
+      <name>setoption()</name>
       <desc>
         <p>One of the
           <seeerl marker="#options">SCTP Socket Options</seeerl>.</p>
       </desc>
     </datatype>
```

Comment:
You need to add `getoption` in the list of documented types as the inet docs now refer to it.

Reply:
Ok yes I see that now, and also the optionval (return from inet:getopts) case needs to be fixed.
Although, I'm not sure what the best way to handle this is.
The type specs in gen_sctp now match what the code does.
But they are not so great for automatically generating the doc.
As I mentioned earlier, just to include `setoption` in the xml results in an expansion like this:
```
setoption() = elementary_option() | record_option()
```
Of course I could include all those types but it feels like that would just bloat the doc without really adding anything. Especially since the same would also apply for `getoption()` and `optionval()`

I'd be interested in what approach you suggest. I assume there is not a way to get the docgen to expand certain types?

Reply:
This is one of the bad parts of having the typespecs as the source for the documentation. Sometimes you need to make compromises either in the type annotations, or the documentation. I'll leave the recommendations about what to do to the people responsible for inet/sctp as I don't really know my way around these parts.

> I assume there is not a way to get the docgen to expand certain types?

Not that I know of.